Fluid.Tests/MiscFiltersTests.cs

```diff
@@ -527,7 +527,8 @@ public async Task DateIsRenderedWithCulture()
         [InlineData("ALLCAPS", "allcaps")]
         [InlineData("One1Two2Three3", "one1-two2-three3")]
         [InlineData("ONE1TWO2THREE3", "one1two2three3")]
-        [InlineData("First_Second_ThirdHi", "first_second_third-hi")]
+        [InlineData("First_Second_ThirdHi", "first-second-third-hi")]
```

Comment:
> The string is converted to lowercase, and any spaces and special characters replaced by hyphens (-).

According to the liquid docs, if the underscore is a special character that means it should be replaced by hypen